File: Dotnet/TauriBackend/EmailNotification.cs
using System.Net;
using System.Net.Mail;
using System.Text.Json;

namespace VRCX.TauriBackend;

public static class EmailNotification
{
    public static async Task<object> SendEmail(JsonElement args)
    {
        var config = JsonSerializer.Deserialize<EmailConfig>(args.GetRawText());
        if (config == null)
            throw new InvalidDataException("Invalid email configuration");

        if (string.IsNullOrWhiteSpace(config.SmtpHost))
            throw new ArgumentException("SMTP host is required");
        if (string.IsNullOrWhiteSpace(config.FromAddress))
            throw new ArgumentException("From address is required");
        if (string.IsNullOrWhiteSpace(config.ToAddress))
            throw new ArgumentException("Recipient address is required");

        using var message = new MailMessage();
        message.From = new MailAddress(config.FromAddress, config.FromName);
        message.To.Add(new MailAddress(config.ToAddress, config.ToName));
        message.Subject = config.Subject ?? string.Empty;
        message.Body = config.Body ?? string.Empty;
        message.IsBodyHtml = config.IsHtml;

        using var client = new SmtpClient(config.SmtpHost, config.SmtpPort)
        {
            EnableSsl = config.SmtpUseSsl,
            Timeout = 30000
        };

        if (!string.IsNullOrWhiteSpace(config.SmtpUsername) &&
            !string.IsNullOrWhiteSpace(config.SmtpPassword))
        {
            client.Credentials = new NetworkCredential(
                config.SmtpUsername,
                config.SmtpPassword
            );
        }

        await client.SendMailAsync(message);
        return true;
    }

    private class EmailConfig
    {
        public string SmtpHost { get; set; } = "";
        public int SmtpPort { get; set; } = 465;
        public bool SmtpUseSsl { get; set; } = true;
        public string SmtpUsername { get; set; } = "";
        public string SmtpPassword { get; set; } = "";
        public string FromAddress { get; set; } = "";
        public string FromName { get; set; } = "VRCX-Pro";
        public string ToAddress { get; set; } = "";
        public string ToName { get; set; } = "";
        public string Subject { get; set; } = "";
        public string Body { get; set; } = "";
        public bool IsHtml { get; set; } = false;
    }
}

File: src/services/emailNotification.js
import { toast } from 'vue-sonner';
import { useEmailNotificationsSettingsStore } from '../stores/settings/emailNotifications';

/**
 * Send an email via the .NET sidecar SMTP handler.
 * @param {object} params
 * @param {string} params.subject
 * @param {string} params.body
 * @param {string} [params.recipientAddress]
 * @param {string} [params.recipientName]
 * @returns {Promise<boolean>}
 */
export async function sendEmail({ subject, body, recipientAddress, recipientName }) {
    const emailStore = useEmailNotificationsSettingsStore();
    const config = emailStore.getConfig();

    if (!config.enabled || !config.smtpHost) {
        console.warn('Email notifications are not configured');
        return false;
    }

    const to = recipientAddress || config.recipientAddress;
    const toName = recipientName || config.recipientName || '';

    if (!to) {
        console.warn('No recipient address configured');
        return false;
    }

    try {
        const result = await AppApi.SendEmail(JSON.stringify({
            smtpHost: config.smtpHost,
            smtpPort: config.smtpPort,
            smtpUseSsl: config.smtpUseSsl,
            smtpUsername: config.smtpUsername,
            smtpPassword: config.smtpPassword,
            fromAddress: config.smtpFromAddress,
            fromName: config.smtpFromName,
            toAddress: to,
            toName,
            subject,
            body,
            isHtml: false
        }));

        if (result === true || result === 'true') {
            return true;
        }
        console.error('Email send failed:', result);
        return false;
    } catch (err) {
        console.error('Email send error:', err);
        toast.error(`Email send failed: ${err.message || err}`);
        return false;
    }
}

/**
 * Send a notification rule email.
 * @param {object} rule - The notification rule that was triggered
 * @param {string} message - The notification message
 * @param {object} eventData - Additional event data
 * @returns {Promise<boolean>}
 */
export async function sendRuleEmail(rule, message, eventData) {
    const emailStore = useEmailNotificationsSettingsStore();
    if (!emailStore.isConfigured()) {
        return false;
    }

    const subject = `[VRCX-Pro] ${rule.name || rule.type}`;
    const body = [
        `Notification Rule: ${rule.name || rule.type}`,
        `Type: ${rule.type}`,
        `Message: ${message}`,
        '',
        eventData ? `Details: ${JSON.stringify(eventData, null, 2)}` : '',
        '',
        `Time: ${new Date().toLocaleString()}`
    ].filter(Boolean).join('\n');

    return sendEmail({ subject, body });
}

/**
 * Send a general notification email.
 * @param {string} title
 * @param {string} message
 * @returns {Promise<boolean>}
 */
export async function sendGeneralEmail(title, message) {
    return sendEmail({
        subject: `[VRCX-Pro] ${title}`,
        body: message
    });
}
